### PR TITLE
Linear computation of the Legendre polynomials

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

File: numerics/combinatorics_test.cpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  static_assert(Binomial(3, 0) == 1, "");
+  static_assert(Binomial(3, 1) == 3, "");
+  static_assert(Binomial(3, 2) == 3, "");
+  static_assert(Binomial(3, 3) == 1, "");
+
+  static_assert(Binomial(4, 0) == 1, "");
+  static_assert(Binomial(4, 1) == 4, "");
+  static_assert(Binomial(4, 2) == 6, "");
+  static_assert(Binomial(4, 3) == 4, "");
+  static_assert(Binomial(4, 4) == 1, "");
```
Comment (eggrobin) [inline numerics/combinatorics_test.cpp]: We use C++17, so we don't empty strings.

File: numerics/combinatorics_test.cpp [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+TEST(Combinatorics, DoubleFactorial) {
+  // https://oeis.org/A000165, https://oeis.org/A001147.
```
Comment (eggrobin) [inline numerics/combinatorics_test.cpp]: http://oeis.org/A006882.

Comment (pleroy) [reply numerics/combinatorics_test.cpp]: Done.

File: numerics/combinatorics_test.cpp (hunk 1 above)
Comment (pleroy) [reply numerics/combinatorics_test.cpp]: TIL.

**Pull request merged**